src/stores/index.js

```diff
@@ -851,23 +896,38 @@ const Store = Reflux.createStore({
     // Set the connection's app name to the electron app name of Compass.
     connection.appname = electron.remote.app.getName();
```

Comment:
this line looks like it quite doesn't belong here, seems more something that should happen somewhere else, or at least after the `!this.state.isConnecting`

Reply:
moved below - agreed though it should be set when we create the connection model to begin with.